core/src/word_lock.rs

```diff
@@ -149,9 +149,12 @@ impl WordLock {
 
             // Get our thread data and prepare it for parking
             state = with_thread_data(|thread_data| {
+                #[cfg(all(unix, not(all(feature = "nightly", target_os = "linux"))))]
```

Comment:
This feels a bit wierd, and is one of the reasons why I left the whole `ThreadParker` API as unsafe in the first place. At least add a comment explaining why this is necessary.

Reply:
Fixed